lib/trino-parquet/src/test/java/io/trino/parquet/reader/TestingColumnReader.java

```diff
@@ -622,6 +629,26 @@ private static Assertion<Number> assertLongTimestampWithTimeZoneNanos(int precis
         };
     }
 
+    private static Assertion<DecodedTimestamp> assertInt96LongTimestampWithTimeZone(int precision)
+    {
+        TimestampWithTimeZoneType type = createTimestampWithTimeZoneType(precision);
+        return (values, block, offset, blockOffset) ->
+        {
+            long epochSeconds = values[offset].epochSeconds();
+            int nanos = values[offset].nanosOfSecond();
+            long epochNanos = epochSeconds * Timestamps.NANOSECONDS_PER_SECOND + nanos;
+            if (precision < 9) {
+                epochNanos = round(epochNanos, 9 - precision);
+            }
```

Comment:
Is this code section tested anywhere?

have `.123456789` in the storage , but having `MICROSECONDS` or `MILLISECONDS` the precision being read?

Reply:
Yes, this is called when TIMESTAMP_TZ_MICROS is tested, `io.trino.parquet.reader.TestingColumnReader#WRITE_INT96` populates nanos which result in rounding to lower precision